wicker/core/datasets.py

```diff
@@ -124,3 +128,74 @@ def __getitem__(self, idx: int) -> Dict[str, Any]:
             self._column_bytes_file_cache.resolve_pointers(row, self.schema()),
             self.schema(),
         )
+
+    def _get_parquet_dir_size(self) -> int:
+        """Get the arrow path and find all the files within, count their bytes
+
+        Returns:
+            int: bytes in parquet directory
+        """
+        # bytes size of arrow table not bytes in arrow table
+        arrow_path = self._s3_path_factory.get_dataset_partition_path(self._partition, s3_prefix=False)
+        bucket, key = arrow_path.replace("s3://", "").split("/", 1)
+
+        def get_folder_size(bucket, prefix):
+            total_size = 0
+            for obj in boto3.resource("s3").Bucket(bucket).objects.filter(Prefix=prefix):
+                total_size += obj.size
+            return total_size
+
+        return get_folder_size(bucket, key)
+
+    def _get_dataset_size(self):
+        """Gets total size of the dataset in bits
+
+        Returns:
+            int: total dataset size in bits
+        """
+        # intialize with size of parquet dir
+        total_bytes = self._get_parquet_dir_size()
+
+        # need to know which columns are heavy pntr columns we go to for
+        # byte adding
+        schema = self.schema()
+        heavy_pointer_cols = []
+        for col_name in schema.get_all_column_names():
+            if schema.get_column(col_name).is_heavy_pointer:
+                heavy_pointer_cols.append(col_name)
+
+        # Each individual row only knows which column file it goes to, so we have to
+        # neccesarily parse all rows :( to get the column files. This should be cached
+        # as metadata but that would require re-curating the datasets.
+        buckets_keys = set()
+        worker = ShuffleWorker(storage=self._storage)
+        for heavy_pntr_col in heavy_pointer_cols:
```

Comment:
Would we benefit from parallelizing this loop?

Reply:
I ended up adding parallelization but I haven't pushed it up yet. So yes it does benefit